### PR TITLE
update `__picKernelArea()`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- change interface of `__picKernelArea()` to support alpaka
- use new interface for kernel calls inside of PIConGPU

### CODE REVIEW

File: src/picongpu/include/simulation_classTypes.hpp [hunk 1]
```diff
@@ -44,33 +44,55 @@ namespace picongpu
...
-#define PIC_PMACC_CUDAKERNELCONFIG(block,...) <<<mapper.getGridDim(),(block),  \
-    __VA_ARGS__+0,                                                             \
-    taskKernel->getCudaStream()>>> PIC_PMACC_CUDAPARAMS
+#define PIC_CUDAKERNELCONFIG(block)                                            \
+    const auto&& gridExtent(mapper.getGridDim());                              \
+    const auto&& blockExtent(block);                                           \
+    PIC_CUDAPARAMS
 
 /**
  * Calls a CUDA kernel and creates an EventTask which represents the kernel.
```
Comment (erikzenker) [inline src/picongpu/include/simulation_classTypes.hpp]: Calls a Alpaka kernel

File: src/picongpu/include/simulation_classTypes.hpp [hunk 2]
```diff
@@ -44,33 +44,55 @@ namespace picongpu
...
  */
-#define __picKernelArea(kernelname,description,area) {                               \
-    CUDA_CHECK_KERNEL_MSG(cudaDeviceSynchronize(),"picKernelArea crash before kernel call");       \
-    AreaMapping<area,MappingDesc> mapper(description);                               \
-    TaskKernel *taskKernel =  Environment<>::get().Factory().createTaskKernel(#kernelname);  \
-    kernelname PIC_PMACC_CUDAKERNELCONFIG
+#define __picKernelArea(description,area, ...) {                               \
+    using KernelType = __VA_ARGS__;                                            \
+    const KernelType theOneAndOnlyKernel;                                      \
+    using KernelDim = ::PMacc::alpaka::Dim<KernelType::kernelDim>;                   \
```
Comment (erikzenker) [inline src/picongpu/include/simulation_classTypes.hpp]: Ahh I see why you need kernelDim from the kernel itself

File: src/picongpu/include/simulation_classTypes.hpp [hunk 3]
```diff
@@ -44,33 +44,55 @@ namespace picongpu
...
 
 /**
  * Calls a CUDA kernel and creates an EventTask which represents the kernel.
  *
- * gridsize for kernel call is set by mapper
- * last argument of kernel call is add by mapper and is the mapper
- *
- * @param kernelname name of the CUDA kernel (can also used with templates etc. myKernnel<1>)
- * @param area area type for which the kernel is called
+ * @param ... name of the CUDA kernel (can also used with templates etc. myKernel<1>)
```
Comment (erikzenker) [inline src/picongpu/include/simulation_classTypes.hpp]: Is this comment still up to date ?

Comment (psychocoderHPC) [inline src/picongpu/include/simulation_classTypes.hpp]: No isn't up to date. I will update this comment

**Comment by psychocoderHPC** [comment]
updated

**Comment by erikzenker** [comment]
thank you

**Pull request merged**